src/qml/imports/Theme/QfToolButtonDrawer.qml

```diff
@@ -0,0 +1,79 @@
+import QtQuick 2.14
+import QtQuick.Controls 2.14
+
+import Theme 1.0
+
+Container {
+  id: container
+
+  property string name: ''
+  property real size: 48
+  property bool downward: true
+  property bool vertical: true
+  property bool collapsed: true
+  property alias round: toggleButton.round
+  property alias bgcolor: toggleButton.bgcolor
+  property alias iconSource: toggleButton.iconSource
+  property alias iconColor: toggleButton.iconColor
+
+  onNameChanged: {
+    collapsed = settings.valueBool("QField/QfToolButtonContainer/"+name+"/collapsed", true)
+  }
+
+  width: vertical
+         ? size
+         : collapsed
+           ? size
+           : size + content.contentWidth + container.spacing * 2
+  height: vertical
+          ? collapsed
+            ? size
+            : size + content.contentHeight + container.spacing * 2
+          : size
+  spacing: 4
+  clip: true
+
+  Behavior on width {
+    enabled: true
+    NumberAnimation { duration: 200 }
+  }
+
+  Behavior on height {
+    enabled: true
+    NumberAnimation { duration: 200 }
+  }
+
+  contentItem: Rectangle {
+    radius: size / 2
+    color: Qt.hsla(toggleButton.bgcolor.hslHue, toggleButton.bgcolor.hslSaturation, toggleButton.bgcolor.hslLightness, 0.3)
+    clip: true
+
+    ListView {
+      id: content
+      width: content.contentWidth
+      height: content.contentHeight
+      x: container.downward ? container.vertical ? container.spacing : container.size + container.spacing : container.spacing
+      y: container.downward ? container.vertical ? container.size + container.spacing : container.spacing : container.spacing
```

Comment:
This is what I meant above with `Direction` enum. While not super complex, if we had a `switch/case` or `if else` instead of boolean combos would be easier.